browser.js

```diff
@@ -190,11 +253,11 @@ Require.loadScript = function (location) {
 Require.ScriptLoader = function (config) {
     var hash = config.packageDescription.hash;
     return function (location, module) {
-        return Promise.fcall(function () {
+        return Promise.try(function () {
 
             // short-cut by predefinition
             if (definitions[hash] && definitions[hash][module.id]) {
-                return definitions[hash][module.id].promise;
+                return definitions[hash][module.id];
```

Comment:
Thanks Kris, all tests pass, we tested quite a few apps as well, what would be the symptoms of preloading breaking, and what could be done to avoid re-creating a defer ?





> On Oct 6, 2015, at 19:12, Kris Kowal <notifications@github.com> wrote:
> 
> In browser.js:
> 
> >  
> >              // short-cut by predefinition
> >              if (definitions[hash] && definitions[hash][module.id]) {
> > -                return definitions[hash][module.id].promise;
> > +                return definitions[hash][module.id];
> This breaks preloading and is related to the Promise.resolve() replacing Promise.defer(). You’ll need to recreate defer().
> 
> function defer() {
>     var deferrred = {};
>     deferred.promise = new Promise(resolve, reject) {
>         deferred.resolve = resolve;
>         deferred.reject = reject;
>     });
>     return deferred;
> }
> —
> Reply to this email directly or view it on GitHub.
>